### DISCUSSION TITLE
meep on macbook pro

### DISCUSSION THREAD

**Discussion by dalarev** [NanoComp/meep#2222] [General]
Hi,

Long-time Windows user here thinking about switching to a macbook pro for everyday use, which will mostly be filled with work on Python, meep, and Julia.

Having never owned a mac before, I'm wondering basic things like: 

- are there compatibility issues? 
- will meep run well on mbp? 
- which mbp would be optimal for heavy meep use (i.e., are the extra few cpu/gpu cores worth it?) 

Bit off topic, but I'm curious to hear meep users' thoughts on this as I've never used macs for scientific computing or technical work.

Thanks!

**Comment by smartalecH** [comment]
Meep works well on Mac (it's Unix-based and easy to build from source if that's what you want). Many users have compiled on the latest Apple silicon. Although I'm pretty sure the conda builds are (currently) only built with Rosetta emulation. 

Of course, meep does not currently have any GPU support, so it will not leverage any of the GPU/neural cores. 

cc @Arjun-Khurana

**Comment by raksharuia** [comment]
Hello @dalarev, I just started to install MEEP on Macbook Pro, and halfway through read that MEEP would work with Scheme interface on Mac OS (https://meep.readthedocs.io/en/latest/Installation/#installation-from-source-on-macos). You mention using python as well. Were you able to compile MEEP to work with python on Mac OS?

**Reply by smartalecH** [reply]
Yes meep with python works on macs (even with Apple silicon) if you build from source.

**Reply by Arjun-Khurana** [reply]
If you only intend to use the python api, you should include the `--without-scheme` flag when you build from source. I had issues with the distribution of Guile on the latest macOS that skipping the scheme installation fixed.

**Reply by dalarev** [reply]
I actually abandoned this effort, @raksharuia. I don't know if this is still true, but I read that, while MEEP can work with Apple silicon, it would be through some emulation as there is still not full compatibility. It may still be faster than a Windows counterpart, but I ended up returning the MBP.

**Reply by raksharuia** [reply]
Thanks, I did try to build from source and installed guile using homebrew (which did not return any specific errors). But at the moment of building libctl or meep, I get errors with guile linking and/or config.

After this, I also tried creating a conda environment using miniconda and installing a pymeep version from conda-forge. Did not succeed. (Here, I think only a rosetta emulation exists and will not native apple).

**Reply by raksharuia** [reply]
@Arjun-Khurana I will try again after eliminating the scheme option (I do not really need it anyway) and see if this time it works. could you please tell me what other changes in the installation process would be required. I have the dependencies installed. I can build harminv too. After this, MEEP documentation suggested - libctl, meep and mpb ... 

I am quite a newbie to building softwares from source and I am not sure if I need to use --without-scheme option for meep or libctl?

**Reply by jjBoca** [reply]
I'm also having trouble building from source on an M1 MBP.  I've followed the steps on the documents page and am unable to build. 

guile is installed via homebrew, but when I try to run configure in libctl, libGDSII, it says that ("/opt/homebrew/opt/pkg-config/bin/pkg-config" "--cflags" "guile-3.0") exited with non-zero error code 127. I was able to install both of those WITHOUT guile and I can see them in /usr/local/ but then I can't configure meep in the same way. 

Please help.

@smartalecH

**Reply by zekishaw** [reply]
There is a bash script somewhere that somebody created that does very well at installing from source and doing all the linking you need on github somewhere. 

I adapted it myself to remove 1/2 small things, I think basically scheme install. 

I can link it to you/drop it in plaintext.

**Reply by zekishaw** [reply]
https://github.com/Dj1312/MacWell/tree/main

This is the script, I think in my case I install the dependancies in conda each time for HDF5 [I can't remember and may be a complete lie] 

You will have to configure your libctl and other bits that look for guile without guile, 

--without-guile should work, and also I reccomend installing with --with-mpi , I doubt you will want to run large parallel simulations on your Mac (it will take ages if you did anyway...) but if you want to run parameter sweeps having it will just help a little bit.

**Reply by jjBoca** [reply]
I still wasn't able to get this to work. Installing with rosetta emulation and Miniconda seems to be the best bet. 

Also, are you saying I'll need to add "--without-guile" to each line in the script? Which other bits look for guile? which line needs --with-mpi?

**Comment by aadame22** [comment]
I keep getting issues when trying to install pymeep on my macbook, im currently getting a 
"Could not load conda plugin `conda-libmamba-solver`:" when running

 'conda create -n mp -c conda-forge pymeep'

ive tried uninstalling and reinstalling as well as using both python 2.7 and the current version of python 

any help would be appreciated, thank you

**Reply by smartalecH** [reply]
This sounds like an issue with your conda install (not necessarily with meep). You might have multiple versions of conda/mamba/micromamba in your path. Might be good to nuke them and start fresh.
